components/req_cpu/src/lib.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+#![feature(shrink_to)]
+
+pub mod collector;
+
+mod future_ext;
+pub use future_ext::FutureExt;
+
+#[cfg(target_os = "linux")]
+mod linux;
+#[cfg(target_os = "linux")]
+pub use linux::{init_recorder, register_collector, CollectorHandle, Guard};
+
+#[cfg(not(target_os = "linux"))]
+mod dummy;
+#[cfg(not(target_os = "linux"))]
+pub use dummy::{install_recorder, register_collector, CollectorHandle, Guard};
+
+use std::sync::Arc;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use collections::HashMap;
+
+pub struct RecorderConfig {
+    record_interval_ms: f64,
+    collect_interval_ms: u64,
+    gc_interval_ms: u64,
+}
+
+impl Default for RecorderConfig {
+    fn default() -> Self {
+        Self {
+            record_interval_ms: 10.1,
+            collect_interval_ms: 1_000,
+            gc_interval_ms: 10 * 60 * 1_000,
+        }
+    }
+}
+
+#[derive(Debug, Default, Eq, PartialEq, Clone, Hash)]
+pub struct RequestTag {
```

Comment:
A better name is preferred. There are too many "tags". How about ResourceMeteringTag